Nonprofit-Exchange-Hub/web-app
# 206feature/file storage provider

## Dev Summary

- File storage service, implements either azure storage for production mode or local disk in development. The service resolves the provider based on the `MODE` env var
- New profile controller with an auth guard to extract the current user from the cookie. Saves the file and stores the url in the user table
- New entity property for the profile url



## Test Plan


repro steps:

1. sign in with postman, and it should store the cookie
2. make a PUT request with multi-form body:
```bash
curl --request PUT \
  --url http://localhost:3001/api/auth/users/profile/12 \
  --header 'Content-Type: multipart/form-data; boundary=---011000010111000001101001' \
  --cookie 'NEH_is_cool=s%253AeyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MTIsImZpcnN0TmFtZSI6InVzZXIxRmlyc3QiLCJsYXN0X25hbWUiOiJ1c2VyMUxhc3QiLCJlbWFpbCI6InVzZXIxRmlyc3RAZXhhbXBsZS5jb20iLCJlbWFpbF9ub3RpZmljYXRpb25fb3B0X291dCI6ZmFsc2UsInByb2ZpbGVfaW1hZ2VfdXJsIjoiaHR0cDovL2xvY2FsaG9zdDozMDAwL3VzZXJwcm9maWxlLTEyLnBuZyIsImlhdCI6MTY3NTk5MjYzMSwiZXhwIjoxNjc1OTk2MjMxfQ.8qVpZq1cZRyLVW2sqEtOTFefGiWJMko1MzTP00SGNOY.KDGi8EYHaQsImli12p2q616NC2%252BgHrwQH2pp3%252FZKO%252BU; .AspNetCore.Antiforgery.4gFydmEKsEU=CfDJ8DlqRFsOpwhFsCNlaNRzxq7LhgKekJCsdzM0QeDsHFUr5l6cMmFPsWnskpl62WtPd_RZ3TxQixaPOUju2w30C67bf54wpc6qs_jgunQtLXN6XEZ467KaoLtWjlbhIUdc6OeikN-0nfd6lNUgkc6Bk4E' \
  --form profile_image_url=@/home/esteban/Downloads/random_avatar.png
```
expected: 

Expect the body:
```json
{
  "id": 12,
  "firstName": "user1First",
  "last_name": "user1Last",
  "email": "user1First@example.com",
  "email_notification_opt_out": false,
  "profile_image_url": "http://localhost:3000/userprofile-12.png"
}
```

## Resources

- Nestjs modules docs, azure blob nest client
